contracts/PathofTheTemplarShard.sol

```diff
@@ -0,0 +1,196 @@
+interface IShards {
+    function partnerMint(
+        address account,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) external;
+}
+/**
+* @title This contract aims to allow a user to mint 
+* an Enclave Shard corresponding to the Enclave quest 
+* upon reaching the winning state of Path of the Temple.
+* It uses EIP712 to verify that a user has signed the hashed message
+*/
+contract PathofTheTemplarShard is 
+    Ownable,
+    AccessControl 
+{
+
+    IShards private SHARDS;
+    IRelic private RELIC;
+
+    bytes32 public constant CHAOS_SHARD_MINTER = keccak256("CHAOS_SHARD_MINTER");
+    bytes32 public constant MYSTERY_SHARD_MINTER = keccak256("MYSTERY_SHARD_MINTER");
+    bytes32 public constant LOGIC_SHARD_MINTER = keccak256("LOGIC_SHARD_MINTER");
+    bytes32 public constant STRUCTURE_SHARD_MINTER = keccak256("STRUCTURE_SHARD_MINTER");  
+    bytes32 public constant ORDER_SHARD_MINTER = keccak256("ORDER_SHARD_MINTER");
+
+    address authorizedMinters;
+    uint256[] public SHARD_ID = [1, 2, 3, 4, 5];
+
+    using Counters for Counters.Counter;
+
+    bytes32 immutable public DOMAIN_SEPARATOR;
+
+    struct QuestCompletedMessageReq {
+        uint256[] enclaves;
+        address signer;
+        address authorizedMinter;
+        uint256 deadline;
+        uint256 nonce;
+    }
+
+    struct EIP712Domain {
+        string name;
+        string version;
+        uint256 chainId;
+    }
+
+    event SignedQuestCompletedMessage(address);
+    event SignedQuestCompletedMessageHash(bytes32 hash);
+
+    error TooManyMessages();
+    error MintForbidden();
+
+    error DeadlineExpired(uint256 lateBy);
+    error InvalidNonce(address account);
+    error InvalidSignature(address account);
+
+    modifier canMint() {
+        if (msg.sender != authorizedMinters) {
+            revert MintForbidden();
+        }
+
+        _;
+    }
+
+mapping(bytes32 => address) signedQuestCompletedMessage;
+mapping(address => uint256[]) public questCompletedMessageBy;
+mapping(address => Counters.Counter) public nonces;
+
+constructor() {
+        authorizedMinters = msg.sender;
+
+        EIP712Domain memory domain = EIP712Domain({
+            name: "PathofTheTemplarShard",
+            version: "1",
+            chainId: 421613
+        });
+        DOMAIN_SEPARATOR = hash(domain);
+    }
+
+function mintPathofthetemplarShard() external canMint {
+    SHARDS.partnerMint(msg.sender, SHARD_ID[0], 1, "");
+}
+
+ function setQuestCompletedMessage(uint256[] calldata signer) external {
+        _setQuestCompletedMessage(msg.sender, signer);
+        emit SignedQuestCompletedMessage(authorizedMinters);
+}
+
+function _setQuestCompletedMessage(address authorizedMinter, uint256[] calldata signer) internal {
+    if (signer.length > 1) {
+        revert TooManyMessages();
+    }
+
+    questCompletedMessageBy[authorizedMinter] = signer;
+}
+
+//This function creates a hashed message for the message signer to confirm their signature
```

Comment:
what's the intended flow from the dapp? you have to call relayedSignQuest...() then call setQuestCompletedMessage()?

could you simplify these function names to be a little less wordy? something like relaySignature() and completeQuest()?

Reply:
I have simplified the functions and also listed a basic flow of intended contract specs in the commentary.
I created a grant Minter Role at the end of the check of signature<>caller address with the relayMintRequest function.